pkg/odo/cli/cli.go

```diff
@@ -152,10 +152,6 @@ func odoRootCmd(name, fullName string) *cobra.Command {
 	_ = pflag.CommandLine.MarkHidden("skip_headers")
 	_ = pflag.CommandLine.MarkHidden("skip_log_headers")
 
-	// We will mark the command as hidden and then re-enable if the command
-	// supports json output
-	_ = pflag.CommandLine.MarkHidden("o")
```

Comment:
By removing this, the option now appears in all commands, event if they do not support it.